test/functional/wallet_sendall.py

```diff
@@ -379,6 +379,60 @@ def sendall_with_maxconf(self):
         assert_equal(len(self.wallet.listunspent()), 1)
         assert_equal(self.wallet.listunspent()[0]['confirmations'], 6)
 
+    @cleanup
+    def sendall_spends_unconfirmed_change(self):
+        self.log.info("Test that sendall spends unconfirmed change")
+        self.add_utxos([17])
+        self.wallet.sendtoaddress(self.remainder_target, 10)
+        assert_greater_than(self.wallet.getbalances()["mine"]["trusted"], 6)
+        self.test_sendall_success(sendall_args = [self.remainder_target])
+
+        assert_equal(self.wallet.getbalance(), 0)
+
+    @cleanup
+    def sendall_spends_unconfirmed_inputs_if_specified(self):
+        self.log.info("Test that sendall spends specified unconfirmed inputs")
+        self.def_wallet.sendtoaddress(self.wallet.getnewaddress(), 17)
+        assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
+        unspent = self.wallet.listunspent(minconf=0)[0]
+
+        self.wallet.sendall(recipients=[self.remainder_target], inputs=[unspent])
+        assert_equal(self.wallet.getbalance(), 0)
+
+    @cleanup
+    def sendall_does_ancestor_aware_funding(self):
+        self.log.info("Test that sendall does ancestor aware funding for unconfirmed inputs")
+
+        # higher parent feerate
+        self.def_wallet.sendtoaddress(address=self.wallet.getnewaddress(), amount=17, fee_rate=20)
+        assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
```

Comment:
The code seems correct to me, but the test seems to have failed on line 408:
https://cirrus-ci.com/task/6362440724643840?logs=ci#L2630

Reply:
`self.def_wallet` is not equal to `self.wallet`, but they belong to the same node. I have implemented this suggestion, but given that the wallets are in the same node, will it make a difference?

Reply:
> `self.def_wallet` is not equal to `self.wallet`, but they belong to the same node. I have implemented this suggestion, but given that the wallets are in the same node, will it make a difference?

Need to call `syncwithvalidationinterfacequeue()` to flush the validation queue before checking the wallet balance. The transaction could have not been processed by the validation interface worker thread which is the one dispatching the tx to the second wallet.